api/tests/test_infractions.py

```diff
@@ -75,3 +76,96 @@ def test_retrieve_single_from_id(self):
 
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json()['id'], self.ban_inactive.id)
+
+
+class CreationTests(APISubdomainTestCase):
+    @classmethod
+    def setUpTestData(cls):  # noqa
+        cls.user = User.objects.create(
+            id=5,
+            name='james',
+            discriminator=1,
+            avatar_hash=None
+        )
+
+    def test_accepts_valid_data(self):
+        url = reverse('bot:infraction-list', host='api')
+        data = {
+            'user': self.user.id,
+            'actor': self.user.id,
+            'type': 'ban',
+            'reason': 'He terk my jerb!',
+            'hidden': True,
+            'expires_at': '5018-11-20T15:52:00+00:00'
+        }
+
+        response = self.client.post(url, data=data)
+        self.assertEqual(response.status_code, 201)
+
+        infraction = Infraction.objects.get(id=1)
+        self.assertEqual(infraction.expires_at, dt.fromisoformat(data['expires_at']))
```

Comment:
`fromisoformat()` is 3.7 only. What's a good alternative to this? The two strings differ in format if I use `isoformat()`.